traveler-ios-ui/TravelerKitUI/ViewControllers/ProductDetailViewController.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  ProductDetailViewController.swift
+//  TravelerKitUI
+//
+//  Created by Omar Padierna on 2019-05-30.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+import TravelerKit
+
+class ProductDetailViewController: UIViewController {
+    public var product:Product?
+
+    private var productDetails:CatalogItemDetails?
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        reload()
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        switch (segue.identifier, segue.destination) {
+        case (_, let vc as BookableProductDetailViewController ):
+            vc.purchasedProduct = product as? BookableProduct
+            vc.productDetails = productDetails
+        case (_, let vc as RetryViewController ):
+            vc.delegate = self
+        case ("loadingSegue", _):
+            break
+        default:
+            Log("Unknown segue", data: nil, level:.warning)
+        }
+    }
+
+    func reload() {
+        guard let product = product else {
+            performSegue(withIdentifier: "errorSegue", sender: nil)
+            return
+        }
+
+        performSegue(withIdentifier: "loadingSegue", sender: nil)
+
+        Traveler.fetchCatalogItemDetails(product, delegate: self)
+    }
+
+    @IBAction func didClose(_ sender: UIBarButtonItem) {
```

Comment:
This is not wrong, but I like to write signatures that are more generic so leaves the storyboard free to use whatever UI element it likes..

What I mean is `sender: Any`